datafusion/functions/src/math/bounds.rs

```diff
@@ -0,0 +1,137 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::datatypes::DataType;
+use datafusion_common::ScalarValue;
+use datafusion_expr::interval_arithmetic::Interval;
+
+fn unary_interval_datatype(interval: &Interval) -> DataType {
+    interval.lower().data_type()
+}
+
+pub(super) fn unbounded(_input: &[&Interval]) -> crate::Result<Interval> {
+    // We cannot assume the input datatype is the same of output type.
+    Interval::make_unbounded(&DataType::Null)
+}
```

Comment:
We actually use the null values within their datatypes for unbounded quantities. Directly using `ScalarValue::Null` is avoided and may cause errors due to type mismatch if the user has not explicitly handled it.